rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/ModelQLAPI.kt

```diff
@@ -0,0 +1,97 @@
+package org.modelix.sample.restapijsonbulk.models.apis
+
+import University.Schedule.N_Lecture
+import University.Schedule.N_Room
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.locations.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.modelix.client.light.LightModelClient
+import org.modelix.sample.restapimodelql.ModelServerLightWrapper
+import org.modelix.sample.restapimodelql.Paths
+import org.modelix.sample.restapimodelql.models.Lecture
+import org.modelix.sample.restapimodelql.models.LectureList
+import org.modelix.sample.restapimodelql.models.Room
+import org.modelix.sample.restapimodelql.models.RoomList
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.net.URLEncoder
+import java.nio.charset.Charset
+
+val logger: Logger = LoggerFactory.getLogger("BulkApi")
+
+object RouteHelper {
+    @JvmStatic
+    public fun urlEncode(input: String): String {
+        return URLEncoder.encode(input, Charset.defaultCharset())!!
+    }
+}
+
+fun Route.ModelQLAPI(mslw: ModelServerLightWrapper) {
+    get<Paths.getLectures> {
+        val allLectures: List<N_Lecture> = mslw.getAllLectures()
+        var lectureList = LectureList()
+        mslw.globalModelClient.runRead {
+            lectureList = LectureList(lectures = allLectures.map { lectureInstance ->
```

Comment:
Doesn't the lambda passed to `runRead` have an output? Having to use a var here is not exactly idiomatic.